runbot_pylint/models/runbot_build.py

```diff
@@ -0,0 +1,217 @@
+            build._log('pylint_script', 'No config file detected')
+            return None
+        path_pylint_conf = os.path\
+            .join(os.path.split(build.server())[0],
+                  build.pylint_conf_path)
+        if not os.path.isfile(path_pylint_conf):
+            build._log('pylint_script', 'No file found [%s]' %
+                       path_pylint_conf)
+            return None
+        modules_to_check_pylint = build.get_modules_to_check_pylint()
+
+        if not modules_to_check_pylint:
+            build._log('pylint_script', 'No modules to check pylint found')
+            return None
+        fname_pylint_run_sh = os.path.join(build.path(),
+                                           'pylint_run.sh')
+        with open(fname_pylint_run_sh, "w") as f_pylint_run_sh:
+            f_pylint_run_sh.write("#!/bin/sh\n")
+            f_pylint_run_sh.write("export PYTHONPATH="
+                                  "$PYTHONPATH:%s\n" %
+                                  (build.server()))
+            for module_to_check_pylint in modules_to_check_pylint:
+                cmd = "pylint --rcfile=%s %s" % \
+                      (path_pylint_conf,
+                       os.path.join(build.server('addons'),
+                                    module_to_check_pylint))
+                f_pylint_run_sh.write(cmd + '\n')
+
+            # TODO: Add check pdb and print sentence check in
+            #       other local script
+            fname_custom_pylint_run = os.path.join(
+                build.path(), "check_ast/check_print_and_pdb.py")
+            if os.path.isfile(fname_custom_pylint_run):
+                for module_to_check_pylint in modules_to_check_pylint:
+                    cmd = "%s %s" % (fname_custom_pylint_run,
+                                     os.path.join(build.server('addons'),
+                                                  module_to_check_pylint))
+                    f_pylint_run_sh.write(cmd + '\n')
+
+        # change mode to execute
+        fpylint_stat = os.stat(fname_pylint_run_sh)
+        os.chmod(fname_pylint_run_sh, fpylint_stat.st_mode | stat.S_IEXEC)
+        return build.spawn([fname_pylint_run_sh],
+                           lock_path, log_path, cpu_limit=2100)
+
+    def job_30_run(self, cr, uid, build, lock_path, log_path):
+        """
+        Inherit method to make logs from pylint errors
+        """
+        res = super(RunbotBuild, self).job_30_run(cr, uid, build, lock_path,
+                                                  log_path)
+        pylint_log = build.path('logs', 'job_15_pylint.txt')
+        pylint_error = False
+        max_log_lines = 20
+        count = 0
+        if os.path.isfile(pylint_log):
+            with open(pylint_log) as fpylint_log:
+                for line in fpylint_log:
+                    if not pylint_error and '****' in line:
+                        pylint_error = True
+                    if pylint_error:
```

Comment:
Yes but the way this is set up is:

1. `pylint_error is False`
2.  `'****' in line is False`
3. both ifs are skipped
4. loop
5. `'****' in line is True`
6. first if is evaluated and `pylint_error is True` 
7. second if is evaluated
8. loop
9. first is never again evaluated because `pylint_error is True`, it doesn't make a difference anyways because the only command is to make it `pylint_error is True` again
10. second if is always run from now on